pkgs/development/compilers/chicken/5/overrides.nix

```
@@ -59,7 +62,9 @@ in
   plot = addToBuildInputs pkgs.plotutils;
   postgresql = addToBuildInputsWithPkgConfig pkgs.postgresql;
   rocksdb = addToBuildInputs pkgs.rocksdb;
-  scheme2c-compatibility = addPkgConfig;
+  scheme2c-compatibility = old:
+    addToNativeBuildInputs (lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [ pkgs.memorymappingHook ])
```

If you're adding a fix for one specific platform only, best just query for that one platform.

```suggestion
    addToNativeBuildInputs (lib.optionals (stdenv.system == "x86_64-darwin") [ pkgs.memorymappingHook ])
```

---

Done.